### ISSUE TITLE
Suggest to include the event source metadata for FillWillRenameEvent

### ISSUE THREAD

**Issue by testforstephen** [microsoft/vscode#114514]
When you rename a primary class name inside a Java file, this will trigger a refactoring to update both this class name and its file name with a workspace edit.
![image](https://user-images.githubusercontent.com/14052197/104886896-6e4dcb00-59a5-11eb-8312-192d453a340f.png)

But when the edit is being applied, it also triggers another FileWillRenameEvent to `workspace.onWillRenameFiles` listener so that the language server recomputes the rename refactoring by mistake. One suggestion is to include the event source metadata to the file event, and then the extension could decide whether to handle it based on the event source. 

```typescript
interface FileWillRenameEvent {
   ...
   eventSource: "applyEdit" | "user";
}
```

**Comment by jrieken** [comment]
Agreed. An alternative is to not send events for edits that are caused by onWill-listeners. We already do something similar for undo-events which never raise an event on the extension host side
